### PR TITLE
Resolve ALIAS/CNAME record before requesting a Kerberos Ticket

### PR DESCRIPTION
**Pull request by k0l0nell**
Using the socket API to attempt to resolve CNAME record for the `proxy_server_address`

### CODE REVIEW

**Comment by k0l0nell** [comment]
For issue #110

File: px.py [hunk 1]
```diff
@@ -681,8 +695,7 @@ def do_proxy_type(self):
                         proxy_auth += header[1] + " "
 
                 for auth in proxy_auth.split():
-                    auth = auth.upper()
-                    if auth in ["NTLM", "KERBEROS", "NEGOTIATE", "BASIC"]:
+                    if auth.upper() in ["NTLM", "KERBEROS", "NEGOTIATE", "BASIC"]:
```
Comment (genotrance) [inline px.py:698]: From other PR?

File: px.py [hunk 2]
```diff
@@ -407,6 +407,20 @@ def __init__(self, proxy_type, proxy_server_address):
                     principal = (urlparse.quote(State.username) + ":" +
                         urlparse.quote(pwd))
 
+            # Attempt to resolve CNAME record for the `proxy_server_address` in case the proxy server is being hosted behind a DNS load balancing mechanism
```
Comment (genotrance) [inline px.py:410]: Might not be the best place to do this since it will happen for every single transaction. Is it possible to do once?

Comment (k0l0nell) [reply px.py:410]: Good point. Once per "proxy refresh" should do I believe. I'll look into it

**Comment by genotrance** [comment]
Closing since no activity - please reopen if you have time to resolve.

**Pull request closed without merging**